go/tasks/pluginmachinery/core/plugin.go

```diff
@@ -52,3 +55,18 @@ type Plugin interface {
 	// Finalize is always called, after Handle or Abort. Finalize should be an idempotent operation
 	Finalize(ctx context.Context, tCtx TaskExecutionContext) error
 }
+
+// Loads and validates a plugin.
+func LoadPlugin(ctx context.Context, iCtx SetupContext, entry PluginEntry) (Plugin, error) {
```

Comment:
This will be called by propeller to load the plugin. We can add more validations here in the future